azure-pipelines/pull-request-validation/build-consumers.yml

```diff
@@ -0,0 +1,196 @@
+# https://docs.microsoft.com/en-us/azure/devops/organizations/accounts/use-personal-access-tokens-to-authenticate
+# Variable: 'commonBranchName' common branch to be used when running the pipeline manually
+name: $(date:yyyyMMdd)$(rev:.r)
+
+trigger: none
+
+variables:
+- name: robolectricSdkVersion
+  value: 28
+
+resources:
+  repositories:
+  - repository: msal
+    type: github
+    name: AzureAD/microsoft-authentication-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: broker
+    type: github
+    name: AzureAD/ad-accounts-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+  - repository: adal
+    type: github
+    name: AzureAD/azure-activedirectory-library-for-android
+    ref: dev
+    endpoint: ANDROID_GITHUB
+
+stages:
+- stage: validateConsumers
+  displayName: Validate Consumer
+  pool:
+    name: Hosted Windows 2019 with VS2019
+  jobs:
+  # setup
+  - job: setupBranch
+    displayName: Setup branch
+    steps:
+    - checkout: none
+    - task: PowerShell@2
+      name: setvarStep
+      displayName: Set var branch
+      inputs:
+        targetType: inline
+        script: |
+          $branch = switch ( '$(Build.Reason)' )
+          {
+              PullRequest
+              {
+                '$(System.PullRequest.SourceBranch)'
+              }
+              Manual
+              {
+                '$(commonBranchName)'
+              }
+              default
+              {
+                ''
+              }
+          }
+          Write-Output "$branch"
+          echo "##vso[task.setvariable variable=commonBranch;isOutput=true]$branch"
+    - powershell: |
+        if ("$(setvarStep.commonBranch)" -eq "") {
+          Write-Output "commonBranchName is not set using default common"
+        }
+        else{
+          Write-Output "$(setvarStep.commonBranch)"
+        }
+      name: echovar
+      displayName: Echo branch name
+  # msal
+  - job: msalValidation
+    displayName: MSAL
+    dependsOn: setupBranch
+    variables:
+      commonBranch: $[ dependencies.setupBranch.outputs['setvarStep.commonBranch'] ]  # map in the variable
+    steps:
+    - template: ../templates/steps/automation-cert.yml
+    - checkout: msal
+      displayName: Checkout msal repository
+      clean: true
+      submodules: recursive
+      persistCredentials: True
+    - task: CmdLine@2
+      displayName: Checkout common submodule $(commonBranch)
+      inputs:
+        script: |
+          git fetch
+          git checkout $(commonBranch)
+          git pull
+          git status
+          git rev-parse HEAD
+        workingDirectory: $(Agent.BuildDirectory)/s/common
+    - task: Gradle@2
+      displayName: Assemble msal
+      inputs:
+        tasks: clean msal:assembleLocal
+    - task: Gradle@2
+      displayName: Run msal Unit tests
+      inputs:
+        tasks: msal:testLocalDebugUnitTest -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}}
```

Comment:
>testLocalDebugUnitTest [](http://example.com/codeflow?start=20&length=22)

for the pipelines.. I will suggest to run tests on release builds or both debug + release instead of just debug.

Reply:
@shahzaibj do you have any comments on this?

Reply:
The test may fail. We will have to run them first and find out. This is because in the past we may have written tests that may have taken a hard dependency on the build type being `debug` which is usually the default build type selected in the developer's IDE. This would primarily be around tests that are potentially testing broker validation as the behavior here used to different between debug and release. However, this has probably now gone way with my changes in https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1191 and we can probably now just do `BrokerValidator.setShouldTrustDebugBrokers(true)` in the tests and that should cover both debug and release scenarios.

I am personally fine either way (testing debug vs release - given that release tests are passing) and if we want to move to testing release that's great but it doesn't necessarily have to be done as part of this PR. (I would let you make the decision)